examples/ci_cd/gitlab-ci-cd/main.tf

```diff
@@ -61,29 +61,19 @@ data "aws_eks_cluster_auth" "cluster" {
 }
 
 locals {
-  tenant      = "aws001"  # AWS account name or unique id for tenant
-  environment = "preprod" # Environment area eg., preprod or prod
-  zone        = "qa"      # Environment with in one sub_tenant or business unit
-
-  kubernetes_version = "1.21"
-
-  vpc_cidr     = "10.2.0.0/16"
-  vpc_name     = join("-", [local.tenant, local.environment, local.zone, "vpc"])
-  cluster_name = join("-", [local.tenant, local.environment, local.zone, "eks"])
-
-  terraform_version = "Terraform v1.0.1"
+  vpc_name     = join("-", [var.tenant, var.environment, var.zone, "vpc"])
+  cluster_name = join("-", [var.tenant, var.environment, var.zone, "eks"])
 }
 
 module "aws_vpc" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "v3.2.0"
+  source = "terraform-aws-modules/vpc/aws"
 
   name = local.vpc_name
-  cidr = local.vpc_cidr
+  cidr = var.vpc_cidr
   azs  = data.aws_availability_zones.available.names
 
-  public_subnets  = [for k, v in slice(data.aws_availability_zones.available.names, 0, 3) : cidrsubnet(local.vpc_cidr, 8, k)]
```

Comment:
we should have the slice back to pickup up to 3 AZs, and if a region doesn't have 3AZs, have the same region `us-west-2` -> we can put a comment explaining about the slice

Reply:
added , Updated code